crates/bitcoin-da/src/service.rs

```diff
@@ -542,21 +590,49 @@ impl DaService for BitcoinService {
 
     #[instrument(level = "trace", skip(self))]
     async fn get_fee_rate(&self) -> Result<u128, Self::Error> {
-        // This already returns ceil, so the conversion should work
-        let res = self.client.estimate_smart_fee().await? as u128;
+        let smart_fee = self.client.estimate_smart_fee(1, None).await?;
+        let btc_vkb = smart_fee.fee_rate.map_or(0.00001f64, |rate| rate.to_btc());
+
+        let sat_vb_ceil = (btc_vkb * 100_000_000.0 / 1000.0).ceil() as u128;
```

Comment:
Seeing this for the second time, maybe separate this into it's own method `estimate_smart_fee`?